### PR TITLE
docs: adds pros n cons for access methods

### PR DESCRIPTION
**Pull request by nilekhc**
Signed-off-by: Nilekh Chaudhari <1626598+nilekhc@users.noreply.github.com>

**Reason for Change**:

This PR adds guidance for access options.

**Requirements**

- [x] squashed commits
- [x] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

fixes: #235 

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

**Comment by nilekhc** [comment]
@aramase Could you suggest some other pros and cons?

**Comment by sozercan** [comment]
Discussed adding initial guidance in https://azure.github.io/secrets-store-csi-driver-provider-azure/configurations/identity-access-modes/

File: website/content/en/configurations/identity-access-modes/_index.md [hunk 1]
```diff
@@ -5,4 +5,13 @@ linkTitle: "Identity Access Modes"
 weight: 1
 description: >
   The Azure Key Vault Provider offers four modes for accessing a Key Vault instance
----
\ No newline at end of file
+---
+
+## Best Practices:
```
Comment (aramase) [inline website/content/en/configurations/identity-access-modes/_index.md:10]:
```suggestion
## Best Practices

```

File: website/content/en/configurations/identity-access-modes/_index.md [hunk 2]
```diff
@@ -5,4 +5,13 @@ linkTitle: "Identity Access Modes"
...
----
\ No newline at end of file
+---
+
+## Best Practices:
+Following order of access modes is recommended for Secret Store CSI driver AKV provider:
+
+| Access Option 	| Comment 	|
+|---	|---	|
+| Pod Identity 	| This is the most secure way to get access to Azure resources (AKV in this case) as it's restricted for a given Pod. 	|
```
Comment (aramase) [inline website/content/en/configurations/identity-access-modes/_index.md:15]:
```suggestion
| Pod Identity 	| This is the most secure way to get access to Azure resources (AKV in this case) as it uses the managed identity bound to the Pod. 	|

```

File: website/content/en/configurations/identity-access-modes/_index.md [hunk 3]
```diff
@@ -5,4 +5,13 @@ linkTitle: "Identity Access Modes"
...
+| Managed Identities (System-assigned and User-assigned) 	| AKS using managed identities manages credentials required to connect to Azure Resources. 	|
```
Comment (aramase) [inline website/content/en/configurations/identity-access-modes/_index.md:16]:
```suggestion
| Managed Identities (System-assigned and User-assigned) 	| Managed identities eliminate the need for developers to manage credentials. Managed identities provide an identity for applications to use when connecting to Azure Keyvault. 	|

nit: managed identities are not specific to aks

```

File: website/content/en/configurations/identity-access-modes/_index.md [hunk 4]
```diff
@@ -5,4 +5,13 @@ linkTitle: "Identity Access Modes"
...
+| Service Principal 	| This is the last option to consider while connecting to AKV as access credentials need to be created as k8s Secret and stored in plain text in etcd.<br>Also, this is the only option to connect to Azure resources from non Azure environment/cluster. 	|
```
Comment (aramase) [inline website/content/en/configurations/identity-access-modes/_index.md:17]:
```suggestion
| Service Principal 	| This is the last option to consider while connecting to AKV as access credentials need to be created as Kubernetes Secret and stored in plain text in etcd.<br>Also, this is the only option to connect to Azure resources from non Azure environment/cluster. 	|
```

**Review by aramase — APPROVED** [review]
lgtm

**Pull request merged**